packages/paste-core/components/switch/src/Switch.tsx

```diff
@@ -1,105 +1,228 @@
 import * as React from 'react';
 import * as PropTypes from 'prop-types';
-import type {SwitchProps} from './types';
+import {useUID} from '@twilio-paste/uid-library';
+import {Text} from '@twilio-paste/text';
+import {Box, safelySpreadBoxProps} from '@twilio-paste/box';
+import type {BoxProps} from '@twilio-paste/box';
 import {SWITCH_HEIGHT, SWITCH_WIDTH} from './constants';
+import {SiblingBox} from '@twilio-paste/sibling-box';
+import {Label} from '@twilio-paste/label';
+import {HelpText} from '@twilio-paste/help-text';
+import {MediaObject, MediaFigure, MediaBody} from '@twilio-paste/media-object';
+import {RequiredDot} from '@twilio-paste/label';
+import {SwitchContext} from './SwitchContext';
 import {SwitchKnob} from './SwitchKnob';
-import {Box, safelySpreadBoxProps} from '@twilio-paste/box';
-import type {BoxStyleProps} from '@twilio-paste/box';
 
-const styles: BoxStyleProps = {
-  backgroundColor: 'colorBackgroundStronger',
-  _checked_hover: {
-    backgroundColor: 'colorBackgroundPrimary',
-    color: 'colorTextLink',
-  },
-  _checked: {
-    backgroundColor: 'colorBackgroundPrimaryStronger',
-    color: 'colorTextLinkStronger',
-  },
-  _hover: {
-    backgroundColor: 'colorBackgroundStrongest',
-    cursor: 'pointer',
-  },
-  _disabled: {
-    backgroundColor: 'colorBackgroundStrong',
-    color: 'colorTextIcon',
-    cursor: 'not-allowed',
-  },
-  _focus: {boxShadow: 'shadowFocus'},
+export interface SwitchProps extends React.InputHTMLAttributes<HTMLInputElement>, Pick<BoxProps, 'element'> {
+  children: NonNullable<React.ReactNode>;
```

Comment:
All good questions, I'll try to answer them one by one:

> we might consider [React.ReactText ](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/1349b640d4d07f40aa7c1c6931f18e3fbf667f3a/types/react/index.d.ts#L236)here since we're expecting a user facing label yea?

Technically people can pass components like icons inside a label, so it isn't strictly text.

> I also am curious if might consider a prop over children using the i18n* prefix to make the translation intent clear? 

The way I'm currently thinking of `i18n` props are that they are a last resort where 'composability' via `children` isn't an option.  Our majority use-case so far is for icon titles embedded within components.

> Finally, since we're extending from BoxProps (which provides its own more permissive type for children) I worry that the union would result in something not as restrictive as we like. We may need to Omit<BoxProps, 'children'> perhaps?

In this instance I believe we're only `Pick`ing `element` from the BoxProps, so it isn't a full extension.